packetery/src/Packetery/Carrier/class-updater.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Packeta carrier updater
+ *
+ * @package Packetery
+ */
+
+declare( strict_types=1 );
+
+namespace Packetery\Carrier;
+
+/**
+ * Class CarrierUpdater.
+ *
+ * @package Packetery
+ */
+class Updater {
+
+	/**
+	 * Carrier repository.
+	 *
+	 * @var Repository
+	 */
+	private $carrier_repository;
+
+	/**
+	 * CarrierUpdater constructor.
+	 *
+	 * @param Repository $carrier_repository Carrier repository.
+	 */
+	public function __construct( Repository $carrier_repository ) {
+		$this->carrier_repository = $carrier_repository;
+	}
+
+	/**
+	 * Validates data from API.
+	 *
+	 * @param array $carriers Data retrieved from API.
+	 *
+	 * @return bool
+	 */
+	public function validate_carrier_data( array $carriers ): bool {
+		foreach ( $carriers as $carrier ) {
+			if ( ! isset(
+				$carrier['id'],
+				$carrier['name'],
+				$carrier['country'],
+				$carrier['currency'],
+				$carrier['pickupPoints'],
+				$carrier['apiAllowed'],
+				$carrier['separateHouseNumber'],
+				$carrier['customsDeclarations'],
+				$carrier['requiresEmail'],
+				$carrier['requiresPhone'],
+				$carrier['requiresSize'],
+				$carrier['disallowsCod'],
+				$carrier['maxWeight']
+			) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Maps input data to storage structure.
+	 *
+	 * @param array $carriers Validated data retrieved from API.
+	 *
+	 * @return array data to store in db
+	 */
+	private function carriers_mapper( array $carriers ): array {
+		$mapped_data = array();
+
+		$carrier_boolean_params = array(
+			'is_pickup_points'         => 'pickupPoints',
+			'has_carrier_direct_label' => 'apiAllowed',
+			'separate_house_number'    => 'separateHouseNumber',
+			'customs_declarations'     => 'customsDeclarations',
+			'requires_email'           => 'requiresEmail',
+			'requires_phone'           => 'requiresPhone',
+			'requires_size'            => 'requiresSize',
+			'disallows_cod'            => 'disallowsCod',
+		);
+
+		foreach ( $carriers as $carrier ) {
+			$carrier_id   = (int) $carrier['id'];
+			$carrier_data = array(
+				'name'       => $carrier['name'],
+				'country'    => $carrier['country'],
+				'currency'   => $carrier['currency'],
+				'max_weight' => (float) $carrier['maxWeight'],
+				'deleted'    => false,
+			);
+			foreach ( $carrier_boolean_params as $column_name => $param_name ) {
+				$carrier_data[ $column_name ] = ( 'true' === $carrier[ $param_name ] );
+			}
+			$mapped_data[ $carrier_id ] = $carrier_data;
+		}
+
+		return $mapped_data;
+	}
+
+	/**
+	 * Saves carriers.
+	 *
+	 * @param array $carriers Validated data retrieved from API.
+	 */
+	public function save_carriers( array $carriers ): void {
```

Comment:
Já osobně bych to poskládal jinak. 

`
$carriers = $mapper->map_carriers($carriers_data); // vrátí pole \Packetery\Carrier\Wrapper (value object)
$repository->save_bulk($carriers);
`

Pokud bych nechtěl injectovat dvě a víc závislostí, tak bych si udělal Facade, která by zjednodušovala používání závislostí.
Když závislosti nejsou do sebe hluboce vnořené, tak se tím snižuje riziko, že se závislosti zacyklí. Mapper nepotřebuje Injektovat repo a repo nepotřebuje injectovat mapper.
Čím víc má třída závislostí tím hůře se testuje.

Reply:
wpdb update a insert potřebují pole, takže mapovat do value objectu nedává teď moc smysl. Nějaký refactoring můžeme udělat po vydání verze 1.